src/components/pages/courses-page/courses-page.tsx

```diff
@@ -44,7 +49,17 @@ const CoursesPage = () => {
     );
   return (
     <div className={styles.wrapper}>
-      <div className={styles.cards}>
+      <div className={styles.header}>
+        <p>Dark mode:</p>
+        <label className={styles.switch}>
+          <input type="checkbox" onChange={() => {
+            localStorage.setItem('theme', theme === 'light' ? 'dark' : 'light')
```

Comment:
Looks good to me!) Only one nitpick, I think, it would be good to separate the theme change logic from the component's rendering code. This way you can keep your component cleaner and more focused on rendering. Maybe, something like this: 
```
const handleThemeChange = () => {
  const newTheme = theme === 'light' ? 'dark' : 'light';
  localStorage.setItem('theme', newTheme);
  dispatch(changeTheme(newTheme));
};
```
and then just call this function on change :)
But this is up to you